### ISSUE TITLE
*Enter your backup passphrase to verify* keyboard does not allow alphabetical characters while I remember and have written such one

### ISSUE THREAD

**Issue by Benjamin-Loison** [Benjamin-Loison/Signal-Android#31]
![image](https://github.com/user-attachments/assets/c4f5bf70-db80-4bad-b112-3e2d77414489)

I guess that *Learn more* redirects to https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#android_enable, see [#issuecomment-2313731192](https://github.com/Benjamin-Loison/Signal-Android/issues/31#issuecomment-2313731192).

![image](https://github.com/user-attachments/assets/591a3758-7348-46ec-a6c5-0257b4ec88de)

Related to #30.

- [ ] I have searched open and closed issues for duplicates
- [ ] I am submitting a bug report for existing functionality that does not work as intended
- [ ] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [ ] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
Describe here the issue that you are experiencing.

### Steps to reproduce
- using hyphens as bullet points
- list the steps
- that reproduce the bug

**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)

### Device info

**Device:** Manufacturer Model XVI
**Android version:** 0.0.0
**Signal version:** 0.0.0

**Comment by Benjamin-Loison** [comment]
No Signal entry in KeePassXC (no interesting entry for *PIN* and *passphrase* too) and only have a single Signal entry in my physical password notebook. Linux Mint 22 Cinnamon Framework 13 only mentions `/home/benjamin/Desktop/bens_folder/softwares/keepassxc/notes.txt` the physical password notebook one, this help ensuring that there is other location in the notebook with the password I am looking for. Could check my other notebook in case I forgot to synchronize them.

```bash
 adb shell input text (CENSORED
```
```
bash: syntax error near unexpected token `('
```

The password starts with `(`.

Other special characters being (with random order and other *bracket*) `:<>+#@[]`.

`'` and `"` around the password do not help and `\` before `(` does not help too.

**Comment by Benjamin-Loison** [comment]
```bash
adb shell input text a
```

does not write `a` nor switches to keyboard.

**Comment by Benjamin-Loison** [comment]
With `scrcpy` keyboard does not work too.

**Comment by Benjamin-Loison** [comment]
Maybe I am confusing with the PIN but what is its purpose then?

<details>
<summary>The person:</summary>

```
-----BEGIN PGP MESSAGE-----

hF4DTQa9Wom5MBgSAQdAVgCEEhlGQlSxf4uNGmmhTivilwDc9FiDrGX02JNpPlMw
DqeBbbv3kmZG/LV6lnmhfhDCFzWQCgOo/kgQi0Wn0qHsTR9gr89BYxs/3rvjkIlR
0kABDX+xDaAvovIgMHiIpk2XQRIwCdjAIloMBY4FTgxx4x1o6cKC+jmYyguDtuSu
oHhKvS/43p5GCh1h/fiTInzh
=qxTa
-----END PGP MESSAGE-----
```
</details>

does not its purpose.

https://support.signal.org/hc/en-us/articles/360007059792-Signal-PIN

> Your Signal PIN is a numeric or alphanumeric code used to help you recover your profile, settings, contacts, and block list if you ever lose or switch devices. It supports features like non-phone number based identifiers. A PIN can also serve as a registration lock.

> A PIN is not a chat backup. Your message history is not linked to a PIN and a PIN cannot be used to recover lost chat history

> If you forget the PIN and have enabled a registration lock, you may be locked out of your account for up to 7 days.

**Comment by Benjamin-Loison** [comment]
```bash
find /home/benjamin/Desktop/bens_folder/devices/phone/ -iname '*signal*'
```

does not return anything interesting on my Linux Mint 22 Cinnamon Framework 13.

```bash
time find ~ -iname '*signal*' 2> /dev/null
```

returns many results.

<details>
<summary>The more precise command:</summary>

```
-----BEGIN PGP MESSAGE-----

hF4DTQa9Wom5MBgSAQdAyRHRx2xfYZsRUxvcbndXY/X57r3NfMwJqZqyekg6v1sw
LJdfv+H+xH1Na6XbWuHv/fufV+UsxUANzL+/hLNUaDRRXMPdmJOSf9Asn/32WysX
0sCKAfDaQhWW50EZY9tCkzAIx1cKDQ6svCqRNAu2MiVVgbWLXGaVv5sq2g9T33nw
dfIT3uLLfBaZsfmnSDbiT0kNTbUBx6tmBffh43ZCAaTfhmxMnZ3C3XwyQa+AxzXh
96iwCMAqVfG61eVg2kAgV/VjiYPNoXWzpCVqS9fqf7IIXkJ2h7K3+pgXhvU131US
os2UpYEdvrJygIs+uKVFz47n0MtAznXX9OTPwBtbJLI8lp4XIAoz9HC4vhdAmVV9
Oi0jVmtYePEIPo2bGmmdvMQKfwQ9MYU0p1cMLsrJmuJU7Qriu9I7GQ5ZjpXeEgXp
DwmYLCY46928VsbEQ3qWThEnt8bUM2ah0b3OvP7Sx9dLt4qilzsFNO8TvHXYPbfU
SiHLxqkcoxuWscfeaLfEEbkouUfmda7CdA/mxr0brm5BVQJDazbGHitag3XF
=Ru/I
-----END PGP MESSAGE-----
```
</details>

shows that there does not seem to be Signal password on Linux Mint 22 Cinnamon Framework 13 as a file like `.txt`.

**Comment by Benjamin-Loison** [comment]
Should check on another device once have a production environment how it looks like.

![image](https://github.com/user-attachments/assets/eedbca56-1051-4cf5-8c68-e10a8fa9af2c)

![image](https://github.com/user-attachments/assets/e5042580-dd8f-4833-9098-00028c744d69)

![image](https://github.com/user-attachments/assets/2f60fa6b-2730-4036-b0a8-bfe523637fc7)

![image](https://github.com/user-attachments/assets/5e85378c-5d2d-4d02-a789-aba1691cf1da)

![image](https://github.com/user-attachments/assets/2d9ba297-d818-4dd5-87a5-3a1799153ebd)

![image](https://github.com/user-attachments/assets/b2360fb4-6155-4fca-b7f5-2afc48c95975)

[Learn more](https://support.signal.org/hc/articles/360007059752)

![image](https://github.com/user-attachments/assets/cda39e22-3ad9-4106-8035-daa28f6483ac)

![image](https://github.com/user-attachments/assets/75ba032c-02f9-47c8-a571-5aca488d12c6)

![image](https://github.com/user-attachments/assets/c21403a0-5dd1-445a-a9bc-a7f81f91af64)

![image](https://github.com/user-attachments/assets/8bee3d88-7cfe-4bd5-9532-8ecf037c6fc5)

![image](https://github.com/user-attachments/assets/27555a99-23bd-48ad-a8eb-144aec894837)

**Comment by Benjamin-Loison** [comment]
In [Benjamin-Loison/android/issues/46#issuecomment-2282301165](https://github.com/Benjamin-Loison/android/issues/46#issuecomment-2282301165) when I restored with SeedVault I only paid attention whether or not Signal has been restored and have not tried actually restoring it:

> Signal has been reinitialized but the 2 backups in `/sdcard/Documents/signal/` seems to have an expected size

While having mentioned the current issue URL, I have not checked that I indeed had this password, as I always make sure to keep things etc.

**Comment by Benjamin-Loison** [comment]
Related to [Benjamin_Loison/seedvault/issues/13](https://codeberg.org/Benjamin_Loison/seedvault/issues/13).

**Comment by Benjamin-Loison** [comment]
![image](https://github.com/user-attachments/assets/b5d48dff-2498-4143-8823-84d6f18ee91d)

![image](https://github.com/user-attachments/assets/bb497cc7-8c9d-4abf-b181-aec930ba9042)

![image](https://github.com/user-attachments/assets/0cdba70d-1665-40b1-9e7e-c4cfdd37f385)

[Learn more](https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages)

https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#android_restore_steps

https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#android_enable

> Record the 30-digit passphrase.

https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#android_forgot

> What if I forget my 30-digit passphrase and need to restore messages on a new phone?

> You will be unable to restore a backup without the passphrase.

So except mentioning that I am looking for a 30-digit passphrase it does not help me.

![image](https://github.com/user-attachments/assets/1efcd887-0be8-454d-a0f2-a8e9616eaf87)

![image](https://github.com/user-attachments/assets/35786fe6-12d5-4c6d-add1-f428f22b4195)

**Comment by Benjamin-Loison** [comment]
Investigate Linux Mint 22 Cinnamon Framework 13 `/home/benjamin/Desktop/bens_folder/devices/phone/backups/11_08_24/sdcard/Pictures/Screenshots/` to make sure I have not taken a screenshot showing the password. I quickly checked all these screenshots and have not noticed anything interesting.

**Comment by Benjamin-Loison** [comment]
Should investigate Signal backup explorers. Well except if the documentation has forgotten to mention that the passphrase type changed, then it should not help.

**Comment by Benjamin-Loison** [comment]
If do not restore, can anyway restore somehow later on even if lose the intermediary new messages? I would say so, as can install offline the APK and restore, see [#issuecomment-2313775258](https://github.com/Benjamin-Loison/Signal-Android/issues/31#issuecomment-2313775258).

Restoring from laptop history would be nice. However, I do not see anything similar in *Signal Desktop Preferences* and https://support.signal.org/hc/en-us/articles/360007059752-Backup-and-Restore-Messages#android_transfer_steps does not seem to apply.

**Comment by Benjamin-Loison** [comment]
iPhone Signal within app settings do not seem to mention backups.

<details>
<summary>At least with the iPhone of the person:</summary>

```
-----BEGIN PGP MESSAGE-----

hF4DTQa9Wom5MBgSAQdAVFkmOI6ZvGHqCi++VQ2J0om/EVhyF192M9nl8wCL7wow
bihqoHqbnqHCNK53OJo+g12htljbj5lHOqnvKUbEJojTCMEvTaOTh2g4M/z6ijPN
0kABv5CCYX7chSFrKjVFIK7qF7oB64zsxJ3kHM9zWUM9wX3BkPuBjkeGdiSxc29o
9izT4iCUODvNZfkvcSbmzQDl
=eBep
-----END PGP MESSAGE-----
```
</details>

**Comment by Benjamin-Loison** [comment]
If click on *Restore* without having typed anything then get:

![image](https://github.com/user-attachments/assets/7e9e0330-0c76-480b-a7e8-e7f0fbe83bbb)

for at least 17 minutes, so I guess that nothing is going on.

If type `0`, then *Restore*, then it shows *Checking...* for a second and then repropose *Restore backup*.

If I restore, then will I lose my laptop Signal history? At least not before enabling Internet access on the phone. See [#issuecomment-2313736427](https://github.com/Benjamin-Loison/Signal-Android/issues/31#issuecomment-2313736427).

**Comment by Benjamin-Loison** [comment]
Alternatively being able to unlock the screen (necessary for decryption) on my bit by bit backup would do the trick, so should not delete it too quickly, see [Benjamin-Loison/android/issues/46#issuecomment-2282301165](https://github.com/Benjamin-Loison/android/issues/46#issuecomment-2282301165). Next to `sda`, I created:
`sda_README_DO_NOT_DELETE.txt`:
```
https://github.com/Benjamin-Loison/Signal-Android/issues/31#issuecomment-2313780646
```

> I decided to get rid of such useless as of right now at least binaries.

**Comment by Benjamin-Loison** [comment]
<details>
<summary>Linux Mint 22 Cinnamon Framework 13 <i>Note to Self</i> goes to 03/03/23 same for discussion with the person:</summary>

```
-----BEGIN PGP MESSAGE-----

hF4DTQa9Wom5MBgSAQdAuESYBVB93h+hWKBaNQYIVLguBp/sIzjrODSC7vhFcxsw
qO7f/89DDOwypM8QWxt/VzTowKbKMdLkNXePa66DL3g3/vIQOywMqtM7/stti/15
0kABwqNkRqifePOz2MsjSv9zH9Cvs99/SSmCjIOUmFc4l530BgyzpcnOuS399FgX
PTXJRtCURdodnJlYJQLVPOWX
=8U0r
-----END PGP MESSAGE-----
```
</details>

To not lose that, let us backup Signal data.

*Signal Desktop Preferences* > *Privacy* > *Delete application data* does not tell where they are stored. Let us stop Signal and verify so with `ps aux | grep -i [s]ignal`. DuckDuckGo search *Where are Signal desktop data stored on Linux?*

```bash
ls ~/.config/Signal
```
<details>
<summary>Output:</summary>

```
 attachments.noindex   DawnWebGPUCache            'Session Storage'
 blob_storage          Dictionaries               'Shared Dictionary'
 Cache                 drafts.noindex              shared_proto_db
'Code Cache'           ephemeral.json              SharedStorage
 config.json           GPUCache                    sql
 Cookies               IndexedDB                   stickers.noindex
 Cookies-journal      'Local Storage'              temp
 Crashpad              logs                       'Trust Tokens'
 databases            'Network Persistent State'  'Trust Tokens-journal'
 DawnCache             optionalResources           VideoDecodeStats
 DawnGraphiteCache     Preferences                 WebStorage
```
</details>

Source: [the Unix Stack Exchange answer 513439](https://unix.stackexchange.com/a/513439)

```bash
du -sh ~/.config/Signal
```
```
3.2G	/home/benjamin/.config/Signal
```

Backup stored at `/home/benjamin/Desktop/bens_folder/devices/framework/Signal/`. I verified that it has the same size.

Long story short, the history of the laptop is kept on re-sync and new messages are correctly sync but there is no transfer of the laptop history to the phone.

In our last Messenger messages, on 04/12/21 I asked this person to continue discussing on Signal.

Could investigate Asus laptop, Pegasus Ubuntu or even Windows setup for further history than my laptop one.

Debian 12 GNOME *Pegasus* *Note to Self* goes to 23/04/24 which helps in the context of #74.

Could also investigate old phone backups (like Fairphone OS and /e/ OS ones if I ever done them). Linux Mint 22 Cinnamon Framework 13 `/home/benjamin/Desktop/bens_folder/devices/phone/` does not seem to contain any other backup.

**Comment by Benjamin-Loison** [comment]
The question is since when do I use Signal? Maybe Messenger can tell me so.

I bought my Fairphone 4 on Feb 18, 2022. So I am mainly looking for my Fairphone 4 Signal data. When have I installed /e/ OS would help.

**Comment by Benjamin-Loison** [comment]
PIN does not seem to somehow encrypt private key on Signal servers:

![image](https://github.com/user-attachments/assets/da40ab86-2afc-4d7e-8a16-baca9fd7611b)

![image](https://github.com/user-attachments/assets/c4c9a91d-c6bf-4bd6-b3c8-ad6f017c90f9)

![image](https://github.com/user-attachments/assets/02982254-a428-4e62-9f57-7b50ab038ba8)

It only seems from other persons sides.

Same issue on Android (Xiaomi Mi A3):

![image](https://github.com/user-attachments/assets/5f467a25-705f-4577-afd4-ef033455f713)

Would be interesting next time suspect that this issue would happen to write the keys that should be identical on both sides as on my sides it seems that as detect others as verified, it should be the same key, so no difference in theory.

**Comment by Benjamin-Loison** [comment]
<details>
<summary>The person:</summary>

```
-----BEGIN PGP MESSAGE-----

hF4DTQa9Wom5MBgSAQdAGBVpQ+N6CQEfCzP2nGPRAfQ9L8JHrcRvUVqagmXFtBUw
1J4TXOxzPYzaAIfKXMSBjmg7ghy3WwzFYDYuKzoc9cbRXXoU/Q5lxrg8rBWA3vbk
0kABCUbxP8ggXLudFAmd6O4eGX0Vj5UquJl79PHFh8xkKa3INncAWzy1v+avHDrc
NzBynau9svNKXBDm7RZ1MHZH
=yyBB
-----END PGP MESSAGE-----
```
</details>

has used his Signal desktop between approximatively *Aug 28, 2024, 1:47 AM GMT+2* and Signal message 1725220170215?

![image](https://github.com/user-attachments/assets/6fd7ca10-4dc1-42c6-83d1-d8f28f323549)

![image](https://github.com/user-attachments/assets/f4eb77b5-d7d7-4154-947c-4671f927cc4b)

![image](https://github.com/user-attachments/assets/86094e8b-5b93-4219-9be0-bdcfa062884a)

![image](https://github.com/user-attachments/assets/11848166-5a86-4f2e-aab9-23e0ec1b7afd)

![image](https://github.com/user-attachments/assets/5501b972-de57-4877-9f70-154072e40dce)

**Comment by Benjamin-Loison** [comment]
Once restored backup:

![image](https://github.com/user-attachments/assets/0e48c76b-6683-4d80-a99f-189b4904ec69)

![image](https://github.com/user-attachments/assets/21c718b7-efb9-4c60-afb3-a2b33d9e975b)

![image](https://github.com/user-attachments/assets/19a36207-8434-4888-94e5-c2f9004803f6)

![image](https://github.com/user-attachments/assets/2a7b8222-2fd7-423a-adbe-7d103371ee86)

![image](https://github.com/user-attachments/assets/384ee5ff-07ca-4992-b2bb-5360b6102451)

![image](https://github.com/user-attachments/assets/ce7c78f6-ca49-4d2d-8d04-dd1a1984f648)

![image](https://github.com/user-attachments/assets/d22ba09d-ce5d-4a1a-abf2-0e8694d976c9)

![image](https://github.com/user-attachments/assets/6ed59746-c498-4e51-93c8-d17ac471809f)

**Comment by Benjamin-Loison** [comment]
Faced same issue as #38.

![image](https://github.com/user-attachments/assets/7faf8251-e741-4864-803b-9e142d044137)

![image](https://github.com/user-attachments/assets/f66f19e1-01bc-4a6c-944b-622ddebe01ff)

![image](https://github.com/user-attachments/assets/dcd15095-7917-436e-a1ac-5e2b28192907)

![image](https://github.com/user-attachments/assets/47dfec03-4f0d-4270-b5c2-0699b777dff6)

If I remember correctly just after an app restart it hides the warning:

![image](https://github.com/user-attachments/assets/ceffef28-0b26-40b3-9e28-9b79b147a792)

**Comment by Benjamin-Loison** [comment]
Linux Mint 22 Cinnamon Framework 13 `/home/benjamin/Desktop/bens_folder/bazaar/memos/computer_science/softwares/signal/` does not help. Maybe I had not this laptop when I started backups.

**Comment by Benjamin-Loison** [comment]
The Signal secure channels might have been lost due to not restoring a backup.

**Comment by Benjamin-Loison** [comment]
Related to #14.

**Comment by Benjamin-Loison** [comment]
Related to [Benjamin-Loison/Signal-Desktop/issues/41](https://github.com/Benjamin-Loison/Signal-Desktop/issues/41).

**Comment by Benjamin-Loison** [comment]
[Benjamin-Loison/Signal-Desktop/issues/41#issuecomment-2374075609](https://github.com/Benjamin-Loison/Signal-Desktop/issues/41#issuecomment-2374075609) may be able to merge histories or just generate a backup from desktop history.

**Comment by Benjamin-Loison** [comment]
Could also leverage the history of my contacts but this is not possible for *Note to Self*.

**Comment by Benjamin-Loison** [comment]
Should propose/enforce to verify the backup passphrase once specified it to the user, to ensure that he remembered it somehow.
